lib/ansible/modules/cloud/linode/linode_v4.py

```diff
@@ -0,0 +1,307 @@
+
+EXAMPLES = """
+- name: Create a new Linode.
+  linode_v4:
+    label: new-linode
+    type: g6-nanode-1
+    region: eu-west
+    image: linode/debian9
+    root_pass: passw0rd
+    authorized_keys:
+      - "ssh-rsa ..."
+    state: present
+
+- name: Delete that new Linode.
+  linode_v4:
+    label: new-linode
+    state: absent
+"""
+
+RETURN = """
+instance:
+  description: The instance description in JSON serialized form.
+  returned: Always.
+  type: dict
+  sample: {
+    "root_pass": "foobar",  # if auto-generated
+    "alerts": {
+      "cpu": 90,
+      "io": 10000,
+      "network_in": 10,
+      "network_out": 10,
+      "transfer_quota": 80
+    },
+    "backups": {
+      "enabled": false,
+      "schedule": {
+        "day": null,
+        "window": null
+      }
+    },
+    "created": "2018-09-26T08:12:33",
+    "group": "Foobar Group",
+    "hypervisor": "kvm",
+    "id": 10480444,
+    "image": "linode/centos7",
+    "ipv4": [
+      "130.132.285.233"
+    ],
+    "ipv6": "2a82:7e00::h03c:46ff:fe04:5cd2/64",
+    "label": "lin-foo",
+    "region": "eu-west",
+    "specs": {
+      "disk": 25600,
+      "memory": 1024,
+      "transfer": 1000,
+      "vcpus": 1
+    },
+    "status": "running",
+    "tags": [],
+    "type": "g6-nanode-1",
+    "updated": "2018-09-26T10:10:14",
+    "watchdog_enabled": true
+  }
+"""
+
+
+from ansible.module_utils.basic import AnsibleModule, env_fallback
+from ansible.module_utils.linode import get_user_agent
+
+try:
+    from linode_api4 import Instance, LinodeClient
+    from linode_api4.errors import (
+        ApiError as LinodeApiError,
+        UnexpectedResponseError
+    )
+    HAS_LINODE_DEPENDENCY = True
+except ImportError:
+    HAS_LINODE_DEPENDENCY = False
+
+
+def create_linode(module, client, **kwargs):
+    """Creates a Linode instance and handles return format."""
+    if kwargs['root_pass'] is None:
+        kwargs.pop('root_pass')
+
+    try:
+        response = client.linode.instance_create(**kwargs)
+    except (LinodeApiError, UnexpectedResponseError) as exception:
+        raise module.fail_json(msg=(
+            'Unable to query the Linode API. Saw: %s'
+        ) % exception)
+
+    try:
+        if isinstance(response, tuple):
+            instance, root_pass = response
+            instance_json = instance._raw_json
+            instance_json.update({'root_pass': root_pass})
+            return instance_json
+        else:
+            return response._raw_json
+    except TypeError:
+        raise module.fail_json(msg=(
+            'Unable to parse Linode instance creation '
+            'response. Please raise a bug against this '
+            'module on https://github.com/ansible/ansible/issues'
+        ))
+
+
+def maybe_instance_from_label(module, client):
+    """Try to retrieve an instance based on a label."""
+    try:
+        label = module.params['label']
+        result = client.linode.instances(Instance.label == label)
+        if len(result) > 1:
```

Comment:
this shouldn't be possible - a Linode customer cannot have two instances with the same label - did you encounter a specific scenario that led you to try to handle this case?

Reply:
Nope, just didn't want to see that bug raised! Great, I'll remove this check then. Thanks!